projects/organization-management/src/app/models/node/node.helper.ts

```diff
@@ -0,0 +1,106 @@
+import { NodeData } from './node.interface';
+import { Node, NodeTree } from './node.model';
+
+export class NodeHelper {
+  static rootsFirst(a: NodeData, b: NodeData): number {
+    if (a.relationships.parentNode && !b.relationships.parentNode) {
+      return -1;
+    }
+    if (!a.relationships.parentNode && b.relationships.parentNode) {
+      return 1;
+    }
+    if (!a.relationships.parentNode && !b.relationships.parentNode) {
+      return 0;
+    }
+    if (a.relationships.parentNode && b.relationships.parentNode) {
+      return 0;
+    }
+  }
+
+  /**
+   * Create a new empty tree with no nodes.
+   */
+  static empty(): NodeTree {
+    return {
+      edges: {},
+      nodes: {},
+      rootIds: [],
+    };
+  }
+
+  static setParent(root: Node, childs: NodeTree): NodeTree {
+    return {
+      edges: { ...childs.edges, [root.id]: childs.rootIds },
+      nodes: { ...childs.nodes, [root.id]: root },
+      rootIds: [root.id],
+    };
+  }
+
+  static merge(current: NodeTree, incoming: NodeTree): NodeTree {
+    if (!current || !incoming) {
+      throw new Error('falsy input');
+    }
+
+    return {
+      edges: NodeHelper.mergeEdges(current.edges, incoming.edges),
+      nodes: NodeHelper.mergeNodes(current.nodes, incoming.nodes),
+      rootIds: NodeHelper.mergeRootIDs(current.rootIds, incoming.rootIds),
+    };
+  }
+
+  private static mergeRootIDs(current: string[], incoming: string[]): string[] {
+    if (incoming && incoming.length > current.length) {
+      return NodeHelper.removeDuplicates([...incoming, ...current]);
+    } else {
+      return NodeHelper.removeDuplicates([...current, ...incoming]);
+    }
```

Comment:
I was a bit confused until I saw where it came from. `removeDuplicates` can be abstracted. Why not even just use [`uniq`](https://lodash.com/docs/4.17.15#uniq)?